### DISCUSSION TITLE
Prevent data from being undefined when re-fetching

### DISCUSSION THREAD

**Discussion by paulintrognon** [vercel/swr#793] [Q&A]
Hi all, I want `data` to stick to its old value when re-fetching, instead of going back to be undefined. I could not find a param to get this behavior. Did I miss something obvious or is it just not an available feature?

Here's what I'd like:

1. first fetch happening, data === undefined
2. first fetch completed, data === [1, 2, 3]
3. second fetch happening, data === [1, 2, 3] (instead of undefined)
4. second fetch completed, data === [4, 5, 6]

Thanks all!

**Comment by gg2001** [comment]
Has anyone find a workaround to allow for this? Can't believe that this isn't documented anywhere, seems like a necessary feature.

**Reply by iamyoki** [reply]
I met that problem too, it's really frustrating, I haven't found any solution yet.

**Comment by ramyareye** [comment]
You might be wrong. useSWR by default caches the value until a new mutation.
